btm/src/main/java/bitronix/tm/internal/XAResourceHolderState.java

```diff
@@ -139,7 +139,35 @@ public boolean isSuspended() {
     public boolean isFailed() {
         return failed;
     }
+    
+    public void quickSuspend() throws XAException {
+        if (!this.started) {
+            throw new BitronixXAException("resource hasn't been started, cannot suspend it: " + this, XAException.XAER_PROTO);
+        }
+        if (this.suspended) {
+            throw new BitronixXAException("resource already suspended: " + this, XAException.XAER_PROTO);
+        }
+
+        if (log.isDebugEnabled()) { log.debug("quick suspending " + this); }
+
+        //take effect
+        this.suspended = true;
+    }
+
+    public void quickResume() throws XAException {
+       if (!this.started) {
```

Comment:
I've changed my approach on quick suspend/remove slightly to leave the resource-holder in a started state.  This seems to be more aligned with the state transitions below (it has not been ended, and start is not expected).